.github/workflows/release-app.yml

```diff
@@ -122,14 +122,13 @@ jobs:
 
       - name: Open changelog PR
         uses: peter-evans/create-pull-request@v5
-        if: "!cancelled()"
```

Comment:
I'm not sure I understand why this was removed -- it sounds like it relates to this part of the description:

> This PR also fixes the changelog PR step to only run if the rest of the workflow completes

Was this check broken and is no longer necessary? Is it related to the critical bug?

Reply:
It isn't part of the critical bug, but it doesn't make sense at all, `!cancelled` would also run if the workflow fails, which you can see in the linked failure. That doesn't make sense: if we skip the changelog when we cancel the release workflow, why would we want it if the release workflow failed? We can close the PR if we don't want the changelog, and we'll need to delete the tag in that case too anyway (and generate a new drafted release when we push whatever fix we need).

Reply:
>  if we skip the changelog when we cancel the release workflow, why would we want it if the release workflow failed? 

Agree with the logic here, although I don't understand how removing this check does what is described in the PR description (actually preventing the step from running if the rest of the workflow hasn't completed).

Reply:
Because `!cancelled()` is strictly _only if the workflow isn't cancelled_, meaning it still runs if the workflow failed. I don't know how this is the case, but you can see it in the linked failure, where a previous step in the workflow fails but the changelog PR step still runs.

Reply:
I understand that and agree it's not useful to only check that case. I assumed (I think wrongly?) that removing this check would mean that now it will run in all cases, not that it would only run if the previous steps completed as described.

Reply:
Ah, I see! The default behaviour is to run only if the previous step succeeded, in effect having `if: success()` by default (https://docs.github.com/en/actions/learn-github-actions/expressions#success).

To run something in every case, you'd have to explicitly use `always`: https://docs.github.com/en/actions/learn-github-actions/expressions#always